app/components/UserValidator/index.tsx

```diff
@@ -38,38 +43,50 @@ const isUser = (user: User | Res): user is User => {
 const Validator = (props: Props) => {
   const { clearSearch, handleSelect, validateAbakusGroup } = props;
   const input = useRef<HTMLInputElement | null | undefined>(null);
-  const [completed, setCompleted] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [scanResults, setScanResults] = useState<ScanResult[]>([]);
   const [showScanner, setShowScanner] = useState(false);
 
-  const showCompleted = () => {
-    setCompleted(true);
-    setTimeout(() => setCompleted(false), 2000);
-  };
+  const addScanResult = (result: string) =>
+    setScanResults((prevScanResults) => {
+      if (
+        prevScanResults.length !== 0 &&
+        prevScanResults[0].message === result
+      ) {
+        prevScanResults[0].count++;
+        return prevScanResults;
+      }
+      return [{ message: result, count: 1 }, ...prevScanResults];
+    });
```

Comment:
Just an idea: Pushing the message whose count was most recently updated to the top.
(Solution: Always remove/slice from list on match, increase count, and then push to list head)

Reply:
I'm not sure that I understand exactly what you mean - or if it's just that the code could be clearer as to what it does.

Currently new scans are added to the start of the list, so that when the list is printed on the website, the most recently scanned item is displayed first, and the older items overflow into nothingness.  

Did you want to make a change to that, or just change the way it is implemented in the code?